moped-editor/src/views/projects/projectView/ProjectStatusBadge.js

```diff
@@ -0,0 +1,242 @@
+    },
+    potential: {
+      color: primary,
+      background: backgroundColors.warning,
+      icon: RemoveCircleOutlineOutlinedIcon,
+    },
+    canceled: {
+      color: white,
+      background: backgroundColors.error,
+      icon: CancelOutlinedIcon,
+    },
+    "on hold": {
+      color: primary,
+      background: backgroundColors.default,
+      icon: PauseCircleOutlineOutlinedIcon,
+    },
+    complete: {
+      color: white,
+      background: backgroundColors.info,
+      icon: CheckCircleOutlineOutlinedIcon,
+    },
+    default: {
+      color: primary,
+      background: defaultBackgroundColor,
+      icon: defaultIcon,
+    },
+  };
+
+  return styleMapping?.[phase] ?? styleMapping.default;
+};
+
+/**
+ * Picks the right color based on the status-phase combination
+ */
+const useFontColorStyles = makeStyles(theme => ({
+  root: {
+    // Find text color
+    color: props => getStyle(theme, (props?.phase ?? "").toLowerCase()).color,
+  },
+}));
+
+/**
+ * Picks the right chip color based on status-phase combination
+ */
+const useChipStyles = makeStyles(theme => ({
+  root: {
+    marginLeft: "1.5rem",
+    fontWeight: "500",
+    fontSize: "16px",
+    borderRadius: "2rem",
+    height: "2.5rem",
+    padding: ".5rem",
+
+    // Find background color
+    backgroundColor: props => getStyle(theme, props?.phase ?? "").background,
+  },
+}));
+
+/**
+ * Renders a chip
+ * @param {number} status - The status id of the current project
+ * @param {string} phase - The current phase name of the project
+ * @param {Object} projectStatuses - A dictionary of all available project statuses
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const ProjectStatusBadge = ({ status, phase, projectStatuses }) => {
+  /**
+   * Returns the label given a status-phase combination
+   * @param {number} status - The status id number
+   * @param {string} phase - The name of the phase
+   * @returns {string}
+   */
+  const getComponentMapName = (status, phase) =>
+    status > 1 // is not active?
+      ? // Then it can be found as a status
+        (
+          projectStatuses.find(s => s.status_id === status)?.status_name ??
+          String(phase)
+        ) // if not, default to phase
+          .toLowerCase()
+      : // if it's active then it's a phase for sure
+        String(phase).toLowerCase();
+
+  /**
+   * Bundle status properties object
+   */
+  const statusProperties = {
+    status: status,
+    phase: getComponentMapName(status, phase), // It's here so it can be corrected by status id
+  };
+
+  /**
+   * Generate font, chip and icon classes
+   */
+  const chipClasses = useChipStyles(statusProperties);
+  const iconClasses = useFontColorStyles(statusProperties);
+
+  // If we don't have a status, then do not render.
+  if (!!!status) return null;
+
+  /**
+   * Create an abstract component pointer
+   */
+  const ChipIcon =
+    getStyle(null, getComponentMapName(status, phase))?.icon ?? defaultIcon;
```

Comment:
Why does theme get passed as `null` here? 

the function definition for getStyle says if theme is passed as null, the function returns null, so shouldn't this always coalesce to defaultIcon?

Reply:
@chiaberry Ah, excellent call. I need to change the docs. That was in fact the behavior until I corrected it. There used to be a statement like this `if(!status) return null;` and that always gave me the default icon no matter what.

Reply:
That function is the most cringe-worthy of the PR, I could not think of a better way to make the component reusable and easy to configure. I'd greatly appreciate any suggestions. 🙏

Reply:
i think it's fine as is, but since you asked...this might be slightly better:

```javascript
const componentMapName = getComponentMapName = (status, phase, projectStatuses);

const statusProperties = {
status: status,
phase: componentMapName, // It's here so it can be corrected by status id
};

const ChipIcon =
    getStyle(null, componentMapName)?.icon ?? defaultIcon;

```